src/som/interpreter/actors/AbstractPromiseResolutionNode.java

```diff
@@ -51,15 +63,19 @@ protected void chainPromise(final SResolver resolver,
       final boolean isBreakpointOnPromiseResolution) {
     assert resolver.assertNotCompleted();
     SPromise promiseToBeResolved = resolver.getPromise();
-
+    boolean breakpointOnResolution = isBreakpointOnPromiseResolution;
     synchronized (promiseValue) {
       Resolution state = promiseValue.getResolutionStateUnsync();
       if (SPromise.isCompleted(state)) {
         resolvePromise(state, resolver, promiseValue.getValueUnsync(),
-           isBreakpointOnPromiseResolution);
+            breakpointOnResolution);
       } else {
         synchronized (promiseToBeResolved) { // TODO: is this really deadlock free?
-          promiseToBeResolved.setTriggerResolutionBreakpointOnUnresolvedChainedPromise(isBreakpointOnPromiseResolution);
+          // TODO: check this condition.
```

Comment:
ok, but what do you want me to check here? Is there anything else that needs to be considered? Why is there a TODO here?